scipy/sparse/linalg/eigen/arpack/arpack.py

```diff
@@ -1231,11 +1231,30 @@ def eigs(A, k=6, M=None, sigma=None, which='LM', v0=None,
             import warnings
             warnings.warn('M does not have the same type precision as A. '
                           'This may adversely affect ARPACK convergence')
+
     n = A.shape[0]
 
-    if k <= 0 or k >= n:
-        raise ValueError("k=%d must be between 1 and ndim(A)-1=%d"
-                         % (k, n - 1))
+    if k <= 0:
+        raise ValueError("k=%d must be greater than 0." % (k, n - 1))
+
+    if k >= n - 1:
+        if issparse(A):
+            raise TypeError("Cannot use scipy.linalg.eig for sparse A with "
+                            "k >= N - 1. Use scipy.linalg.eig(A.toarray()) or"
+                            " reduce k.")
+
+        import warnings
+        warnings.warn("k >= N - 1 for N * N square matrix. "
+                      "Using scipy.linalg.eig instead.", UserWarning)
+
+        if isinstance(A, LinearOperator):
+            raise TypeError("Cannot use scipy.linalg.eig for LinearOperator "
+                            "A with k >= N - 1.")
+        if isinstance(M, LinearOperator):
+            raise TypeError("Cannot use scipy.linalg.eig for LinearOperator "
+                            "M with k >= N - 1.")
```

Comment:
This `LinearOperator` checks need to happen before the warning.

Reply:
I think we should inform the user that we are using or trying to use `scipy.linalg.eig` right after the check `k>=N`(or `k>=N-1`). Or maybe we can put it in the docs somewhere.

Reply:
Ok, I see what you're saying here. (Sorry for the long wait!) Can you move the warning up before the `TypeError` checks here? And maybe have it say "Attempting to use" instead of "Using"?